src/main/scala/ch/epfl/bluebrain/nexus/iam/routes/EventRoutes.scala

```diff
@@ -0,0 +1,136 @@
+package ch.epfl.bluebrain.nexus.iam.routes
+
+import java.util.UUID
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.javadsl.server.Rejections._
+import akka.http.scaladsl.marshalling.sse.EventStreamMarshalling._
+import akka.http.scaladsl.model.headers.`Last-Event-ID`
+import akka.http.scaladsl.model.sse.ServerSentEvent
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.{Directive1, PathMatcher0, Route}
+import akka.persistence.query._
+import akka.persistence.query.scaladsl.EventsByTagQuery
+import akka.stream.scaladsl.Source
+import ch.epfl.bluebrain.nexus.iam.acls.AclEvent.JsonLd._
+import ch.epfl.bluebrain.nexus.iam.acls.{AclEvent, Acls}
+import ch.epfl.bluebrain.nexus.iam.config.AppConfig
+import ch.epfl.bluebrain.nexus.iam.config.AppConfig.tracing._
+import ch.epfl.bluebrain.nexus.iam.config.AppConfig.{HttpConfig, PersistenceConfig}
+import ch.epfl.bluebrain.nexus.iam.directives.AuthDirectives._
+import ch.epfl.bluebrain.nexus.iam.io.TaggingAdapter._
+import ch.epfl.bluebrain.nexus.iam.permissions.PermissionsEvent
+import ch.epfl.bluebrain.nexus.iam.permissions.PermissionsEvent.JsonLd._
+import ch.epfl.bluebrain.nexus.iam.realms.RealmEvent.JsonLd._
+import ch.epfl.bluebrain.nexus.iam.realms.{RealmEvent, Realms}
+import ch.epfl.bluebrain.nexus.iam.routes.EventRoutes._
+import ch.epfl.bluebrain.nexus.iam.types.{Caller, Permission}
+import ch.epfl.bluebrain.nexus.iam.{acls => aclsp, permissions => permissionsp, realms => realmsp}
+import io.circe.Encoder
+import io.circe.syntax._
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+import scala.util.{Failure, Success, Try}
+
+/**
+  * The event stream routes.
+  *
+  * @param acls   the acls api
+  * @param realms the realms api
+  */
+class EventRoutes(acls: Acls[Task], realms: Realms[Task])(implicit as: ActorSystem,
+                                                          hc: HttpConfig,
+                                                          pc: PersistenceConfig) {
+
+  private val pq: EventsByTagQuery = PersistenceQuery(as).readJournalFor[EventsByTagQuery](pc.queryJournalPlugin)
+
+  private implicit val implAcls: Acls[Task] = acls
+
+  def routes: Route = {
+    (handleRejections(RejectionHandling()) & handleExceptions(ExceptionHandling()) & pathPrefix(hc.prefix)) {
+      concat(
+        routesFor("acls" / "events", permissionsEventTag, aclsp.write, typedEventToSse[AclEvent]),
```

Comment:
aclEventTag ?

Reply:
Also add a TODO for `aclsp.write`. It should be, in a following PR changes to `acls/read`. Same for listing ACLs and fetching ACLs